entrypoint.sh

```diff
@@ -51,27 +51,51 @@ case ${RUNNER_SCOPE} in
     ;;
 esac
 
-if [[ -n "${ACCESS_TOKEN}" ]]; then
-  _TOKEN=$(bash /token.sh)
-  RUNNER_TOKEN=$(echo "${_TOKEN}" | jq -r .token)
+# If the variable is not set, set it with the default value
+if [ -z "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}" ]; then
+  CONFIGURED_ACTIONS_RUNNER_FILES_DIR="/actions-runner-files"
 fi
 
-echo "Configuring"
-./config.sh \
-    --url "${_SHORT_URL}" \
-    --token "${RUNNER_TOKEN}" \
-    --name "${_RUNNER_NAME}" \
-    --work "${_RUNNER_WORKDIR}" \
-    --labels "${_LABELS}" \
-    --runnergroup "${_RUNNER_GROUP}" \
-    --unattended \
-    --replace
+# Loading the files from the mounted directory
+if [ -d "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}" ]; then
+  cp -p -r "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}/." "/actions-runner"
+fi
 
-unset RUNNER_TOKEN
+if [ -f "/actions-runner/.runner" ]; then
+  echo "The runner has already been configured"
+else
+
+    if [[ -n "${ACCESS_TOKEN}" ]]; then
+      echo "Obtaining the token of the runnet"
+      _TOKEN=$(bash /token.sh)
+      RUNNER_TOKEN=$(echo "${_TOKEN}" | jq -r .token)
+    fi
+
+    echo "Configuring"
+    ./config.sh \
+        --url "${_SHORT_URL}" \
+        --token "${RUNNER_TOKEN}" \
+        --name "${_RUNNER_NAME}" \
+        --work "${_RUNNER_WORKDIR}" \
+        --labels "${_LABELS}" \
+        --runnergroup "${_RUNNER_GROUP}" \
+        --unattended \
+        --replace
+
+    # Saving the files in another directory for the possibility to mount them from the host next time
+    if [ -d "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}" ]; then
+      # Quoting (even with double-quotes) the regexp brokes the copying
+      cp -p -r "/actions-runner/_diag" "/actions-runner/svc.sh" /actions-runner/.[^.]* "${CONFIGURED_ACTIONS_RUNNER_FILES_DIR}"
+    fi
+
+    if [[ ${_DISABLE_AUTOMATIC_DEREGISTRATION} == "false" ]]; then
+      trap deregister_runner SIGINT SIGQUIT SIGTERM
+    fi
 
-if [[ ${_DISABLE_AUTOMATIC_DEREGISTRATION} == "false" ]]; then
-  trap deregister_runner SIGINT SIGQUIT SIGTERM
 fi
 
```

Comment:
we still want to keep the `_DISABLE_AUTOMATIC_DEREGISTRATION` logic here outside of the if else

Reply:
Yeah, I changed it locally and wait for you comments in order to decide whether to merge your PR or push only this change

Reply:
I commented your PR. As we finish with the additional envvar discussion, I will merge your PR